here-naksha-lib-handlers/src/main/java/com/here/naksha/lib/handlers/SourceIdHandler.java

```diff
@@ -80,17 +77,31 @@ protected EventProcessingStrategy processingStrategyFor(IEvent event) {
     return event.sendUpstream(request);
   }
 
-  private void transformPropertyOperation(ReadFeatures readRequest) {
+  /**
+   * For the AND case, any property query sub-clause that can be converted into tag query will be converted and returned, leaving the remaining inconvertible clauses intact.
+   * <br>
+   * For the OR case, it is required that every sub-clause must be convertible, else the whole clause will not be converted.
+   * This is because OR relation between types of queries (property, tag, spatial,...) is not supported, only AND is supported and applied at the very end of the request.
+   */
+  public static void mapIntoTagOperation(ReadFeatures readRequest) {
 
     if (readRequest.getQuery().getProperties() == null) {
       return;
     }
 
     IPropertyQuery propertyOp = readRequest.getQuery().getProperties();
 
-    // TODO fix it CASL-710
-    //    PropertyOperationUtil.transformPropertyInPropertyOperationTree(
-    //            propertyOp, SourceIdHandler::mapIntoTagOperation);
+    final Optional<ITagQuery> tagQuery = transformPropertyOperation(propertyOp);
+    if (tagQuery.isPresent()) {
+      readRequest.getQuery().setTags(tagQuery.get());
```

Comment:
By doing so, we will lose (override) the original `tagQuery` that was in `ReadFeatures`. 
Is that intended?

If not (or we simply don't know) then we could to the following:
1) Check if there are any tag queries in the original request
2a) If there is a tag query, combine it via `AND` with tag query translated from property. We can utiliza `TagAnd` for that.
2b) If there is no tag query in the original request, go with the current approach (set just translated property)

Reply:
But first, let's try to establish if tag query and property query can be both present here.
My intuition is that it is totally possible but TBH I did not analyze the flow

Reply:
That's true, I missed this case. Yes there could be already existing tag query which should be kept. Will push the fix like you outlined.